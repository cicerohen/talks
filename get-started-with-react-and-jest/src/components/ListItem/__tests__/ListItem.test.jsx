import React from 'react';
import { shallow } from 'enzyme';
import ListItem from '../ListItem';

const minimalProps = {
  children: <div>list item</div>,
};

const commonProps = {
  ...minimalProps,
  onClick: jest.fn(),
};

describe('<ListItem /> Tests', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render without exploding', (done) => {
    const wrapper = shallow(
      <ListItem {...minimalProps}/>
    );
    expect(wrapper).toHaveLength(1);
    done();
  });

  it('should render correctly', (done) => {
    const wrapper = shallow(
      <ListItem {...minimalProps}/>
    );
    expect(wrapper).toMatchSnapshot();
    done();
  });

  it('should render correctly with all props', (done) => {
    const wrapper = shallow(
      <ListItem {...commonProps} />
    );
    expect(wrapper).toMatchSnapshot();
    done();
  });

  it('should call onClick handler when click on list item', (done) => {
    const wrapper = shallow(<ListItem {...commonProps}/>);
    wrapper.simulate('click');
    expect(commonProps.onClick.mock.calls.length).toBe(1);
    done();
  });
});
