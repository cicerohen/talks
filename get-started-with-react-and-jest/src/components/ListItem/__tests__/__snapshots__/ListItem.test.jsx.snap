// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> Tests should render correctly 1`] = `
<li
  className="root "
>
  <div
    className="container"
  >
    <div>
      list item
    </div>
  </div>
</li>
`;

exports[`<ListItem /> Tests should render correctly with all props 1`] = `
<li
  className="root clickable"
  onClick={[MockFunction]}
>
  <div
    className="container"
  >
    <div>
      list item
    </div>
  </div>
</li>
`;
